mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/configuration/impl/MongoDBConfiguration.java

```diff
@@ -82,14 +86,62 @@ public MongoDBConfiguration(ConfigurationPropertyReader propertyReader, OptionsC
 	 * @return the {@link MongoClientOptions} corresponding to the {@link MongoDBConfiguration}
 	 */
 	public MongoClientOptions buildOptions() {
-		MongoClientOptions.Builder optionsBuilder = new MongoClientOptions.Builder();
+		MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder();
 
 		optionsBuilder.connectTimeout( timeout );
 		optionsBuilder.writeConcern( writeConcern );
 		optionsBuilder.readPreference( readPreference );
+
+		Map<String, Method> settingsMap = createSettingsMap();
+		for ( Map.Entry<String, Method> entry : settingsMap.entrySet() ) {
+			String setting = MongoDBProperties.MONGO_DRIVER_SETTINGS_PREFIX + "." + entry.getKey();
+			// we know that there is exactly one parameter
+			Class<?> type = entry.getValue().getParameterTypes()[0];
+
+			// for reflection purposes we need to deal with wrapper classes
+			if ( int.class.equals( type ) ) {
+				type = Integer.class;
+			}
+			if ( boolean.class.equals( type ) ) {
+				type = Boolean.class;
+			}
+
+			Object property = propertyReader.property( setting, type ).withDefault( null ).getValue();
+			if ( property == null ) {
+				continue;
+			}
+
+			Method settingMethod = entry.getValue();
+			try {
+				settingMethod.invoke( optionsBuilder, property );
+			}
+			catch ( InvocationTargetException | IllegalAccessException e ) {
+				// TODO OGM-810 fix error message
+			}
+		}
+
 		return optionsBuilder.build();
 	}
 
+	private Map<String, Method> createSettingsMap() {
+		Map<String, Method> settingsMap = new HashMap<>();
+
+		Method[] methods = MongoClientOptions.Builder.class.getDeclaredMethods();
+		for ( Method method : methods ) {
+			if ( method.getParameterTypes().length == 1 ) {
+				Class<?> parameterType = method.getParameterTypes()[0];
+				// we just care of string, int and boolean setters
+				if ( String.class.equals( parameterType )
+						|| int.class.equals( parameterType )
```

Comment:
I guess. Depends when you want to merge. I am off next week. I might find some
time somewhere, but not sure. If you want to merge asap, a follow up issue might
be better'

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pi&cv=6.0.64&pv=8.1.3]
On Sat, Aug 1, 2015 at 11:12, hibernate/hibernate-ogm
<reply@reply.github.com>
wrote:
In 
mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/configuration/impl/MongoDBConfiguration.java
[https://github.com/hibernate/hibernate-ogm/pull/572#discussion_r36032088] :

>  		return optionsBuilder.build();
>  	}
>  
> +	private Map<String, Method> createSettingsMap() {
> +		Map<String, Method> settingsMap = new HashMap<>();
> +
> +		Method[] methods = MongoClientOptions.Builder.class.getDeclaredMethods();
> +		for ( Method method : methods ) {
> +			if ( method.getParameterTypes().length == 1 ) {
> +				Class<?> parameterType = method.getParameterTypes()[0];
> +				// we just care of string, int and boolean setters
> +				if ( String.class.equals( parameterType )
> +						|| int.class.equals( parameterType )


@hferentschik [https://github.com/hferentschik] You think you could do that as part of this PR? If not, I'll open another
follow-up just for that one, as it can be done separately from (and much quicker
than) addressing the deprecation issues.

—
Reply to this email directly or view it on GitHub
[https://github.com/hibernate/hibernate-ogm/pull/572/files#r36032088] .[https://github.com/notifications/beacon/AASFoslQVis7S5Ox7Khsgzscgsu1p5a3ks5ojIUTgaJpZM4FjGq_.gif]